sdks/python/apache_beam/io/iobase.py

```diff
@@ -1424,11 +1440,12 @@ def current_progress(self):
     def current_restriction(self):
       start_pos = self._delegate_range_tracker.start_position()
       stop_pos = self._delegate_range_tracker.stop_position()
-      return SourceBundle(
-          self._weight,
-          self._source,
-          start_pos,
-          stop_pos)
+      return _SDFBoundedSourceWrapper._SDFBoundedSourceRestriction(
```

Comment:
Rather than re-creating it here, how about just storing (and returning) the initial _SDFBoundedSourceRestriction object (which does lazy initialization in range_tracker())?